concat-kdf/tests/tests.rs

```diff
@@ -0,0 +1,200 @@
+    let tests = [
+        Test {
+            secret: &hex!("00"),
+            other_info: &[],
+            expected_key: &hex!(
+                "63e1c62226df5825c32eceb32e9f318316f54d8b56a12f764c88ec4f4b4ed80c
+                 b4fd20c2b9fd522a9439f067dfabc8cc2da781dd7b0abc0821909dff25ff33db
+                 bac4619b1569765a23fd8fb91fbad73903d48a1088ae1b710437f2fc08815625
+                 31248ae2f6c12a7156eb27e881ed186f4de45cf22a97cca0d67a8dc5dc7acf8c
+                 6dc447966ff13da6f4d347c7421208f8553a9aeda6f66bcbfc556f47a402f20e
+                 b6f1a0c21bd375fa2246f7faad19a05c22536770ffb3466e544b0723005cea89
+                 40706c1e16e81000d2829e7ab2e1a13f278396da38bd17414cf14d5dfe33212f
+                 c2ca53bb60df0e36fc76d4861e540da48db328b56e9bf5561a30ecc15184e70b"
+            ),
+        },
+        Test {
+            secret: &hex!("00"),
+            other_info: &hex!("00"),
+            expected_key: &hex!(
+                "5a91d2ac619076c5eb6aae55512841d76b6442f98e23372660e854dc5b327844
+                 77e504f2fd620c9328711b682b8a21a0173cfd0c42a96263006e992d8483193f
+                 03347c4cd479f8d0797fe6613fa999cd40bac7fad61715210c044dc722e8db8c
+                 8b4a35076d17b0eb6170429ba2fb51371ddfc6189cb90dc27ae701523a709f67
+                 08c5d1d4140e1434aeb8b593e41f21616543c7727dffe14f4c424dd487d01a92
+                 1370993c359c2a45c9a6584df5a09e36c14453df46a9f9fa8b65ebd790eec122
+                 a0789cabb04854e392656eade6e40d821826708ba1fb078fd1d15565902c68c5
+                 531c0be6415f9b24721fdc118fafaef6f1b5db03d895e7ff503ddbd73d111f38"
+            ),
+        },
+        Test {
+            secret: &hex!("ba5eba11bedabb1ebe5077edb0a710adb01dfacecab005eca11ab1eca55e77e011"),
+            other_info: &hex!("f005ba1100ddba11"),
+            expected_key: &hex!(
+                "247e5d73b8f45ff573294437d6e4fddecf95dee5f54b9ad8db11aa3f856e02b2
+                 5da6159680c87f629c0ef6fbad79ef6499561d444106a765a7c7a96055ee684e
+                 37e406b52ca3c6ca6690b5b93f8b1a5609a4b159a90a8ce9391f6e2738447034
+                 8c2dc71f559cd2a9f471063a056ab138634679587dd51ac6d4729b7f94b44343
+                 00dee4eb7c43c11919f83ee52aa11150c38476b8e675e3d2a20dac03b7153d05
+                 92041e8324b7509f64a098060e74eb7d51ccfd9231154ef0fc0ee9de46252794
+                 9f149b14636338fb9fef85c56f8a5d6f95e9268ed312053abe7cb5f8e6e405a4
+                 058d5ef840e5ec6218ec2dfc4acdff1bea1a5341b9ea47e61642c7214b305083"
+            ),
+        },
+    ];
+
+    test_input_output::<Sha512>(&tests);
+}
+
+#[test]
+fn test_errors() {
+    assert_eq!(
+        concat_kdf::derive_key_into::<Sha512>(&[], &[], &mut [0u8; 42]),
+        Err(concat_kdf::Error::NoSecret)
+    );
+
+    assert_eq!(
+        concat_kdf::derive_key_into::<Sha512>(&[0u8; 42], &[], &mut [0u8; 0]),
+        Err(concat_kdf::Error::ZeroOutput)
+    );
+}
+
+#[test]
+#[ignore = "The test requires huge memory allocation."]
```

Comment:
For testing purposes, it is possible to create a fake slice with the desired length using
https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html

Reply:
Thanks!